From: Ryan Lee <ryan.lee@canonical.com>
Date: Tue, 19 Nov 2024 19:18:33 -0800
Subject: [PATCH 07/13] Set parser/network.h:ip_conds ptrs to null in its
 free_conds after freeing

Picked from MR: https://gitlab.com/apparmor/apparmor/-/merge_requests/1870
Signed-off-by: Ryan Lee <ryan.lee@canonical.com>
(cherry picked from commit a1a1e6ef2d76893f09cd3eb4cebfdd0d9c7d7256)
Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 parser/network.h | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/parser/network.h b/parser/network.h
index 4f429d9..30906ae 100644
--- a/parser/network.h
+++ b/parser/network.h
@@ -142,12 +142,18 @@ public:
 	bool is_none = false;
 
 	void free_conds() {
-		if (sip)
+		if (sip) {
 			free(sip);
-		if (sport)
+			sip = NULL;
+		}
+		if (sport) {
 			free(sport);
-		if (iface)
+			sport = NULL;
+		}
+		if (iface) {
 			free(iface);
+			iface = NULL;
+		}
 	}
 };
 
